core/src/main/scala/org/apache/spark/SparkConf.scala

```diff
@@ -543,6 +543,30 @@ class SparkConf(loadDefaults: Boolean) extends Cloneable with Logging with Seria
       }
     }
 
+    if (contains("spark.cores.max")) {
```

Comment:
I think these checks for negative numbers are redundant with arg checking for `spark-submit`?